File: mpi/hello.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char** argv) {
   int rank, size, ierr;

   ierr = MPI_Init(&argc, &argv);
   ierr = MPI_Comm_size(MPI_COMM_WORLD, &size);
   ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank);

   printf("Hello World, I am rank %d of %d procs\n", rank, size);
   ierr = MPI_Finalize();
   return 0;
}

File: mpi/hello.f90
program mpitest
   use MPI
   integer :: rank, size, ierr, plen
   character*(MPI_MAX_PROCESSOR_NAME) :: pname

   call MPI_Init(ierr)
   call MPI_Comm_size(MPI_COMM_WORLD, size, ierr)
   call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)
   call MPI_Get_processor_name(pname, plen, ierr)

   write(*,'("Hello World, from rank", I5, " of ", I5, " procs")', &
         ADVANCE="NO") rank, size
   write(*,'(", host ", A)') trim(pname)
   call MPI_Finalize(ierr)
end
